rotkehlchen/globaldb/upgrades/v2_v3.py

```diff
@@ -0,0 +1,361 @@
+    """Avoid querying coingecko for now"""
+    # TODO ASSETS: Remove this hack
+    addr = '0x' + "".join(random.choices('0123456789ABCDEFabcdef', k=40))
+    return {k.value: addr for k in CHAINID}
+
+
+def query_coingecko_for_addresses(identifier: str) -> Dict[str, str]:  # noqa: E501 pylint: disable=unused-argument
+    """
+    Call coingecko API to obtain asset addresses in other chains.
+    Can raise:
+    - RemoteError
+    - KeyError
+    """
+    # Hack this function for now since we need this mapping for the migration
+    """
+    url = 'https://api.coingecko.com/api/v3/coins/{coin_id}?tickers=false&market_data=false&developer_data=true&sparkline=false'  # noqa: E501
+    req = requests.get(url.format(coin_id=identifier))
+    data = req.json()
+    output = {}
+    for platform, addr in data['platforms'].items():
+        if len(addr):
+            output[CHAINID.deserialize_from_coingecko(platform).value] = to_checksum_address(addr)
+    return output
+    """
+    return coingecko_hack()
+
+
+def upgrade_ethereum_assets(query: List[Any]) -> EVM_TUPLES_CREATION_TYPE:
+    old_ethereum_data = []
+    old_ethereum_id_to_new = {}
+    evm_tuples = []
+    assets_tuple = []
+    common_asset_details = []
+
+    for entry in query:
+        new_id = evm_address_to_identifier(
+            address=entry[0],
+            chain=CHAINID.ETHEREUM_CHAIN_IDENTIFIER,
+            token_type=EvmTokenKind.ERC20,
+            collectible_id=None,
+        )
+        old_ethereum_id_to_new[entry[3]] = new_id
+        old_ethereum_data.append((new_id, *entry))
+
+    for entry in old_ethereum_data:
+        evm_tuples.append((
+            entry[0],  # identifier
+            EvmTokenKind.ERC20.value,  # token type
+            CHAINID.ETHEREUM_CHAIN_IDENTIFIER.value,  # chain
+            entry[1],  # address
+            entry[2],  # decimals
+            entry[3],  # protocol
+        ))
+        new_swapped_for = old_ethereum_id_to_new.get(entry[8], entry[8])
+        assets_tuple.append((
+            entry[0],  # identifier
+            'C',  # type
+            entry[7],  # started
+            new_swapped_for,  # swapped for
+            entry[0],  # common_details_id
+        ))
+        common_asset_details.append((
+            entry[0],  # identifier
+            entry[5],  # name
+            entry[6],  # symbol
+            entry[9],  # coingecko
+            entry[10],  # cryptocompare
+            None,  # forked
+        ))
+
+    return (
+        evm_tuples,
+        assets_tuple,
+        common_asset_details,
+    )
+
+
+def evm_compatible_assets_create_tuples(
+    query: List[Any],
+) -> EVM_TUPLES_CREATION_TYPE:
+    """Get information from globaldb and transform it to evm compatible structure"""
+    old_id_to_new = {}
+    evm_tuples = []
+    assets_tuple = []
+    common_asset_details = []
+
+    for entry in query:
+        try:
+            ids = query_coingecko_for_addresses(entry[6])
```

Comment:
Querying any external service, especially in a for loop should not be part of a DB upgrade, ever. A DB upgrade is one of the most sensitive things we can do and if it goes wrong the user's installation may be fucked. So we need to keep it as simple as possible.

Reply:
yeah. This logic has to move but it will be in another PR. They ids and information would come from the packaged globaldb but at this point I needed something to simulate this and test that the migration works. Also will be useful to move it into the script we will use to create the new globaldb